src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -55,10 +55,15 @@ export const META_DEFAULTS = {
   hideActionMenu: true
 }
 
-export const openFolder = folderId => {
+export const openFolder = (
+  folderId,
+  openFolderAction = OPEN_FOLDER,
```

Comment:
why not use another param for the action (something like `context` for example) with a default value instead of using different action types? If you use action type you have to add the type in every `switch()` on type whereas using a distinct param, you could treat special cases only when it's necessary

Reply:
In fact this is precisely the goal.  I want to be able to do what I want for a special action without worrying about the consequence because the scoped is controlled. 

The goal is to have in a few weeks / months a clean reducer, cleaned Controllers and cleaned Action. 

I thought about this solution, but I didn't want to add complexity to the reducer and do not want having to do something like : 

```jsx
case OPEN_FOLDER:
if(action.payload.params === 'trash'){
return {
...state,
view: trash
}
}else {
return {
...state,
view: folder
}
}
```

Do you see any value to do that ? Except the `you have to switch on the new action in the reducer`